File: ddpoliglot_v6_flutter/lib/models/data/user_lesson_word.dart
// ignore_for_file: file_names
import 'package:ddpoliglot_v6_flutter/models/dict_word.dart';
import 'package:json_annotation/json_annotation.dart';

import 'Word.dart';

part 'user_lesson_word.g.dart';

@JsonSerializable()
class UserLessonWord {
  int userLessonWordID;
  int userLessonID;
  int wordID;
  Word? word;
  int wordType;
  DictWord? dictWord;

  UserLessonWord(this.userLessonWordID, this.userLessonID, this.wordID,
      this.word, this.wordType);

  factory UserLessonWord.fromJson(Map<String, dynamic> data) =>
      _$UserLessonWordFromJson(data);

  Map<String, dynamic> toJson() => _$UserLessonWordToJson(this);
}

File: ddPoliglotV6.Data/Models/UserLessonWord.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ddPoliglotV6.Data.Models
{
    public class UserLessonWord
    {
        [Key]
        public int UserLessonWordID { get; set; }
        public int UserLessonID { get; set; }
        public int WordID { get; set; }
        public Word Word { get; set; }
        public int wordType { get; set; } // 0 - main, 1 - in repetition list
    }
}
